src/main/scala/io/iohk/ethereum/network/PeerActor.scala

```diff
@@ -71,29 +80,81 @@ class PeerActor(nodeInfo: NodeInfo) extends Actor with ActorLogging {
+        val statusTimeout = system.scheduler.scheduleOnce(waitForStatusInterval, self, StatusReceiveTimeout)
+        context become waitingForNodeStatus(rlpxConnection, statusTimeout)
+      } else {
+        log.info("Connected peer does not support eth {} protocol. Disconnecting.", Message.PV63.toByte)
+        disconnectFromPeer(rlpxConnection, Disconnect.Reasons.IncompatibleP2pProtocolVersion)
+      }
+  }
 
-        rlpxConnection ! RLPxConnectionHandler.SendMessage(
-          CommonMessages.Status(
-            protocolVersion = Message.PV63,
-            networkId = 1,
-            totalDifficulty = new BigInteger("17179869184"),
-            bestHash = ByteString(Hex.decode("d4e56740f876aef8c010b86a40d5f56745a118d0906a34e69aec8c0db1cb8fa3")),
-            genesisHash = ByteString(Hex.decode("d4e56740f876aef8c010b86a40d5f56745a118d0906a34e69aec8c0db1cb8fa3"))))
+  def waitingForNodeStatus(rlpxConnection: ActorRef, timeout: Cancellable): Receive = handleTerminated orElse
+    handlePing(rlpxConnection) orElse {
+    case MessageReceived(status: Status) =>
+      timeout.cancel()
+      log.info("Peer {} returned status ({})", peerId, status)
+      rlpxConnection ! RLPxConnectionHandler.SendMessage(
+        GetBlockHeaders(Left(DaoBlockNumber), maxHeaders = 1, skip = 0, reverse = 0)
+      )
+      val waitingForDaoTimeout = system.scheduler.scheduleOnce(waitForStatusInterval, self, NoDaoHeaderReceived)
+      context become waitingForChainForkCheck(rlpxConnection, status, waitingForDaoTimeout)
+    case StatusReceiveTimeout =>
+      log.warning("Timeout while waiting for peer {} status", peerId)
+      disconnectFromPeer(rlpxConnection, Disconnect.Reasons.TimeoutOnReceivingAMessage)
+    case MessageReceived(d: Disconnect) =>
+      log.info("Received {} from peer {}. Closing connection", d, peerId)
+      context stop self
+  }
 
+  def waitingForChainForkCheck(rlpxConnection: ActorRef, status: Status, timeout: Cancellable): Receive = handleTerminated orElse
+    handlePing(rlpxConnection) orElse {
+    case MessageReceived(msg@BlockHeaders(blockHeader +: Nil)) if blockHeader.number == DaoBlockNumber =>
+      timeout.cancel()
+      log.info("DAO Fork header received from peer {} - {}", peerId, Hex.toHexString(blockHeader.hash))
+      if (daoForkValidator.validate(msg).isEmpty) {
+        log.warning("Peer {} is running the ETC chain", peerId)
         context become new HandshakedHandler(rlpxConnection).receive
       } else {
-        log.info("Peer does not support eth {} protocol. Disconnecting.", Message.PV63.toByte)
-        rlpxConnection ! SendMessage(Disconnect(Disconnect.Reasons.IncompatibleP2pProtocolVersion))
-        context.system.scheduler.scheduleOnce(5.seconds, self, PoisonPill)
+        log.warning("Peer {} is not running the ETC fork, disconnecting", peerId)
+        disconnectFromPeer(rlpxConnection, Disconnect.Reasons.UselessPeer)
+      }
+    case MessageReceived(d: Disconnect) =>
```

Comment:
this is used in quite a few places, maybe we could extract it to `handleDisconnect`. And maybe we could name such message handlers differently, like handleDisconnectMsg, handlePingMsg? Just a suggestion